src/main/java/org/jasig/portlet/widget/links/PortletXmlGroupService.java

```diff
@@ -0,0 +1,72 @@
+package org.jasig.portlet.widget.links;
+
+import java.net.URL;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.annotation.PostConstruct;
+import javax.servlet.ServletContext;
+
+import org.dom4j.Document;
+import org.dom4j.Node;
+import org.dom4j.XPath;
+import org.dom4j.io.SAXReader;
+import org.jaxen.NamespaceContext;
+import org.jaxen.SimpleNamespaceContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class PortletXmlGroupService {
+    private static final String PORTLET_XML_PATH = "/WEB-INF/portlet.xml";
+
+    private Logger log = LoggerFactory.getLogger(this.getClass());
```

Comment:
📓 Lombok provides a nice helper to normalize log usage
Ref: https://projectlombok.org/features/log